### PR TITLE
Rspec for sign up and sign in

### PR DESCRIPTION
**Pull request by josephelias-rubyians**

### CODE REVIEW

File: app/controllers/users/registrations_controller.rb [hunk 1]
```diff
@@ -5,6 +5,7 @@ class Users::RegistrationsController < Devise::RegistrationsController
   def create
     @user = User.new(configure_sign_up_params)  
     if @user.save
+      sign_up(resource_name, resource)
       render json: @user
```
Comment (neethu-satheesh) [inline app/controllers/users/registrations_controller.rb:9]: @josephelias-rubyians  Hi Joseph, lets use the serialiser in this object.

File: spec/controllers/users/sessions_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
+require 'rails_helper'
+
+describe Users::SessionsController, type: :request do
+
+  let (:user) { create_user }
+  let (:login_url) { '/login' }
+  let (:logout_url) { '/logout' }
+
+  context 'When logging in' do
```
Comment (neethu-satheesh) [inline spec/controllers/users/sessions_controller_spec.rb:9]: @josephelias-rubyians  Hi, lets also check for the response data like email, etc

File: spec/controllers/users/registrations_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,50 @@
+require 'rails_helper'
+
+describe Users::RegistrationsController, type: :request do
```
Comment (neethu-satheesh) [inline spec/controllers/users/registrations_controller_spec.rb:3]: @josephelias-rubyians  Good rspec tests. Congrats on writing your first rspec tests 👍   💯

File: spec/controllers/users/sessions_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    it 'returns a token' do
+      expect(response.headers['Authorization']).to be_present
+    end
+
+    it 'returns 200' do
+      expect(response.status).to eq(200)
+    end
+  end
+
+  context 'When password is missing' do
```
Comment (neethu-satheesh) [inline spec/controllers/users/sessions_controller_spec.rb:23]: @josephelias-rubyians  Lets also write a spec for incorrect password

File: spec/controllers/users/registrations_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+      post signup_url, params: {
+        user: {
+          email: existing_user.email,
+          password: existing_user.password
+        }
+      }
+    end
+
+    it 'returns 422' do
+      expect(response.status).to eq(422)
```
Comment (neethu-satheesh) [inline spec/controllers/users/registrations_controller_spec.rb:46]: @josephelias-rubyians  Lets also return a message in json response something like user already exists.

**Review by neethu-satheesh — CHANGES_REQUESTED** [review]

File: spec/controllers/users/registrations_controller_spec.rb (hunk 5 above)
Comment (josephelias-rubyians) [reply spec/controllers/users/registrations_controller_spec.rb:46]: yes devise is returning message like "Email already taken"

File: spec/spec_helper.rb [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+# Given that it is always loaded, you are encouraged to keep this file as
+# light-weight as possible. Requiring heavyweight dependencies from this file
+# will add to the boot time of your test suite on EVERY test run, even for an
+# individual file that may not need all of that loaded. Instead, consider making
+# a separate helper file that requires the additional dependencies and performs
+# the additional setup, and require it from the spec files that actually need
+# it.
+#
+# See http://rubydoc.info/gems/rspec-core/RSpec/Core/Configuration
+RSpec.configure do |config|
```
Comment (neethu-satheesh) [inline spec/spec_helper.rb:16]: @josephelias-rubyians  Something to explore for increasing the efficiency of tests https://github.com/DatabaseCleaner/database_cleaner

File: app/controllers/users/registrations_controller.rb (hunk 1 above)
Comment (josephelias-rubyians) [reply app/controllers/users/registrations_controller.rb:9]: I didn't understand this. I think  sign_up(resource_name, resource) is a helper method for sign up and generating token.

Comment (neethu-satheesh) [reply app/controllers/users/registrations_controller.rb:9]: @josephelias-rubyians  Sorry, I added the comment for line 9. render json: @user

File: spec/controllers/users/registrations_controller_spec.rb (hunk 5 above)
Comment (neethu-satheesh) [reply spec/controllers/users/registrations_controller_spec.rb:46]: okay. Lets write a spec for the error response messages.

**Pull request merged**